File: src-tauri/src/logger.rs
use colored::*;
use serde_repr::{Deserialize_repr, Serialize_repr};

#[derive(Serialize_repr, Deserialize_repr, Debug)]
#[repr(u8)]
pub enum LogLevel {
    Info,
    Warn,
    Error,
    Debug,
    Query,
}

#[tauri::command]
pub async fn log(message: String, level: LogLevel) {
    let colored_message = match level {
        LogLevel::Info => message.green(),
        LogLevel::Warn => message.yellow(),
        LogLevel::Error => message.red(),
        LogLevel::Debug => message.blue(),
        LogLevel::Query => message.magenta(),
    };

    let log_level = match level {
        LogLevel::Info => "INFO",
        LogLevel::Warn => "WARN",
        LogLevel::Error => "ERROR",
        LogLevel::Debug => "DEBUG",
        LogLevel::Query => "QUERY",
    };

    println!("[{}] {}", log_level, colored_message);
}

File: src/helpers/Logger.ts
import { invoke } from "@tauri-apps/api";
import { LogLevels } from "../enums/LogLevels";

export default class Logger {
  static log(level: LogLevels, message: string) {
    // if (process.env.LOG === 'false') return
    // if (process.env.LOG_LEVEL && level < parseInt(process.env.LOG_LEVEL)) return

    invoke("log", {
      level,
      message,
    });
  }

  static error(message: string) {
    Logger.log(LogLevels.error, message);
  }

  static info(message: string) {
    this.log(LogLevels.info, message);
  }

  static debug(message: string) {
    this.log(LogLevels.debug, message);
  }

  static warn(message: string) {
    this.log(LogLevels.warn, message);
  }
}
